### PR TITLE
Actually enable TLS/SSL for old clusters by default

### PR DESCRIPTION
**Pull request by JelteF**
DESCRIPTION: Enable TLS by default for clusters created before Citus 8.1.0

Before this commit clusters created with a version lower than 8.1.0
would not have SSL enabled by default. This changes that to enable it by
default always. The only difference is that old clusters (created before
8.1.0), will have "sslmode=prefer" in their `citus.node_conninfo`
setting. New clusters will have "sslmode=require". These settings are
equivalent from an attackers point of view. They both protect against
passive eavesdropping, but are vulnerable to an active MITM attack.

### CODE REVIEW

File: src/backend/distributed/sql/citus--9.2-2--9.2-3.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+-- Enable SSL to encrypt all trafic by default
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/sql/citus--9.2-2--9.2-3.sql]: s/trafic/traffic

File: src/backend/distributed/utils/enable_ssl.c [hunk 2]
```diff
@@ -73,36 +72,39 @@ citus_setup_ssl(PG_FUNCTION_ARGS)
...
+		Node *citusSSLCiphersParseTree = ParseTreeNode(SET_CITUS_SSL_CIPHERS_QUERY);
+		AlterSystemSetConfigFile((AlterSystemStmt *) citusSSLCiphersParseTree);
 	}
+
+	/*
+	 * ssl=on requires that a key and certificate are present, since we have
+	 * enabled ssl mode here chances are the user didn't install credentials already.
+	 *
+	 * This function will check if they are available and if not it will generate a
+	 * self singed certificate.
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/utils/enable_ssl.c]: s/singed/signed

File: src/backend/distributed/utils/enable_ssl.c [hunk 3]
```diff
@@ -73,36 +72,39 @@ citus_setup_ssl(PG_FUNCTION_ARGS)
...
 		 */
-		CreateCertificatesWhenNeeded();
-
-		GloballyReloadConfig();
+		Node *citusSSLCiphersParseTree = ParseTreeNode(SET_CITUS_SSL_CIPHERS_QUERY);
+		AlterSystemSetConfigFile((AlterSystemStmt *) citusSSLCiphersParseTree);
 	}
+
+	/*
+	 * ssl=on requires that a key and certificate are present, since we have
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/utils/enable_ssl.c]: I guess this is private key, so maybe `.. that a server private key and server certificate are present ...`

File: src/backend/distributed/sql/citus--7.0-1.sql [hunk 4]
```diff
@@ -5,10 +5,21 @@
...
--- create temporary UDF that has the power to change settings within postgres and drop it
--- after ssl has been setup.
+-- Enable SSL to encrypt all traffic by default
+
+-- create temporary UDF that has the power to change settings within postgres
+-- and drop it after ssl has been setup.
+
+-- This was placed here in version 8.1.0. This means that this code has not
+-- been executed for clusters created before 8.1.0. This is a somewhat hacky
+-- way to detect that we can safely set sslmode=require for citus.node_conninfo
```
Comment (JelteF) [inline src/backend/distributed/sql/citus--7.0-1.sql]: If it should change default `sslmode=require` to `sslmode=prefer`

**Comment by onderkalaci** [comment]
> DESCRIPTION: Enable TLS by default for clusters created before Citus 8.1.0


Seems too old, I'm closing the PR. If you think otherwise, please re-open @JelteF

**Pull request closed without merging**